### PR TITLE
Document new build script options [skip ci].

### PR DESCRIPTION
**Pull request by coryan**
I forgot to document the options here in the last couple of changes.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-cpp/1712)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -184,6 +184,8 @@ which combinations are tested regularly.
    documentation.
  * `BUILD_TYPE`: if set, override the `CMAKE_BUILD_TYPE` flag when configuring
    the build.
+ * `BUILD_TESTING`: if set to `no` disable the unit tests, this is sometimes
```
Comment (mbrukman) [inline CONTRIBUTING.md]: Comma after `no`, please.

Start a new sentence with "This is sometimes".

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -209,3 +211,6 @@ which combinations are tested regularly.
    for the Google Cloud C++ Libraries.  Check the
    [configuration file](.clang-tidy) for details on what `clang-tidy` options
    we use.
+ * `CREATE_GRAPHVIZ`: if set to `yes` use `CMake` to generate a dependency graph
```
Comment (mbrukman) [inline CONTRIBUTING.md]: Comma after `yes`, please.

File: CONTRIBUTING.md (hunk 1 above)
Comment (coryan) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 2 above)
Comment (coryan) [reply CONTRIBUTING.md]: Done.

**Comment by coryan** [comment]
@mbrukman PTAL.

**Pull request merged**